magefile.go

```diff
@@ -114,11 +139,234 @@ func (Build) All() {
 	)
 }
 
+func (Build) BuildImages() {
+	mg.Deps(
+		mg.F(Build.imageBuild, "addon-operator-manager"),
+		mg.F(Build.imageBuild, "addon-operator-webhook"),
+		mg.F(Build.imageBuild, "api-mock"),
+		mg.F(Build.imageBuild, "addon-operator-index"), // also pushes bundle
+	)
+}
+
+func (Build) PushImages() {
+	mg.Deps(
+		mg.F(Build.imagePush, "addon-operator-manager"),
+		mg.F(Build.imagePush, "addon-operator-webhook"),
+		mg.F(Build.imagePush, "addon-operator-index"), // also pushes bundle
+	)
+}
+
 // Builds the docgen internal tool
 func (Build) Docgen() {
 	mg.Deps(mg.F(Build.cmd, "docgen", "", ""))
 }
 
+func (b Build) imageBuild(cmd string) error {
```

Comment:
Let's expose this target so that we can have more granular image-building targets in makefile?

imo that's better than calling `build-images` (which builds all the images) everytime we want to build a certain image. For example, the targets like `load-addon-operator`, `load-addon-operator-webhook`, `load-api-mock` redundantly build all the images.

I get it that redundant builds would be subsequently faster because they'll be cached but still, it seems like an unnecessary overhead.

wdyt

Reply:
I forgot that mage can handle parameters so nicely. This Function is now exposed and used from the Makefile to only build images once:
```
build-image-%:
	./mage build:imagebuild $*

push-image-%: registry-login
	./mage build:imagepush $*
```

Reply:
Thanks, it looks great now :)